### PR TITLE
fix unRecurs - see #2548

### PR DESCRIPTION
**Pull request by yllen**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #2548

### CODE REVIEW

File: inc/commondbtm.class.php [hunk 1]
```diff
@@ -1964,9 +1964,15 @@ function canUnrecurs() {
       }
       // TODO : do we need to check all relations in $RELATION["_virtual_device"] for this item
 
+      // check connections of a computer
+      $connectcomputer = array('Monitor', 'Peripheral', 'Phone', 'Printer');
+      if (in_array($this->getType(), $connectcomputer)) {
+         return Computer_Item::canUnrecursSpecif($this, $entities);
```
Comment (remicollet) [inline inc/commondbtm.class.php]: $this is an instantiated object.

File: inc/computer_item.class.php [hunk 2]
```diff
@@ -872,5 +872,41 @@ static function cloneItem($itemtype, $oldid, $newid) {
...
+    * @param  $item        item linked to the computer to check
+    * $param  $entities
+    *
+    * @return boolean
+   **/
+   static function canUnrecursSpecif($item, $entities) {
+      global $DB, $CFG_GLPI;
+
+      // RELATION : computers -> items
+      $object = new $item();
```
Comment (remicollet) [inline inc/computer_item.class.php]: not needed, $item is an object (see previous

File: inc/computer_item.class.php [hunk 3]
```diff
@@ -872,5 +872,41 @@ static function cloneItem($itemtype, $oldid, $newid) {
...
+      $table = getTableForItemType($object->getType());
```
Comment (remicollet) [inline inc/computer_item.class.php]: not used

File: inc/computer_item.class.php [hunk 4]
```diff
@@ -872,5 +872,41 @@ static function cloneItem($itemtype, $oldid, $newid) {
...
+
+      $sql = "SELECT `itemtype`, GROUP_CONCAT(DISTINCT `items_id`) AS ids, `computers_id`
+              FROM `glpi_computers_items`
+              WHERE `glpi_computers_items`.`itemtype` = '".$object->getType()."'
```
Comment (remicollet) [inline inc/computer_item.class.php]: $item->getType();

File: inc/computer_item.class.php [hunk 5]
```diff
@@ -872,5 +872,41 @@ static function cloneItem($itemtype, $oldid, $newid) {
...
+                    AND `glpi_computers_items`.`items_id` = ".$item->fields['id']."
+              GROUP BY `itemtype`";
+      $res = $DB->query($sql);
+
+      if ($res) {
+         while ($data = $DB->fetch_assoc($res)) {
+            $comp = new Computer();
+            if ($comp->isEntityAssign()) {
```
Comment (remicollet) [inline inc/computer_item.class.php]: always true ?

File: inc/computer_item.class.php (hunk 2 above)
Comment (yllen) [inline inc/computer_item.class.php]: OK

File: inc/computer_item.class.php (hunk 3 above)
Comment (yllen) [inline inc/computer_item.class.php]: forget to delete it. OK

File: inc/computer_item.class.php (hunk 4 above)
Comment (yllen) [inline inc/computer_item.class.php]: Of course with previous

File: inc/computer_item.class.php (hunk 5 above)
Comment (yllen) [inline inc/computer_item.class.php]: Yes for a computer.

**Comment by yllen** [comment]
Master done https://github.com/glpi-project/glpi/commit/468199faf643315534277c01bf612a080fb94c69

**Pull request merged**